packages/wallet/src/redux/sagas/message-listener.ts

```diff
@@ -149,8 +150,13 @@ function* validateAgainstLatestCommitment(incomingCommitment: Commitment) {
 function* validateTransition(fromCommitment: Commitment, toCommitment: Commitment) {
   const validTransition = yield contractUtils.validateTransition(fromCommitment, toCommitment);
   if (!validTransition) {
-    const fromAppAttributes = appAttributesFromBytes(fromCommitment.appAttributes);
-    const toAppAttributes = appAttributesFromBytes(toCommitment.appAttributes);
+    const isConsensusChannel = toCommitment.channel.channelType === CONSENSUS_LIBRARY_ADDRESS;
+    const fromAppAttributes = isConsensusChannel
+      ? appAttributesFromBytes(fromCommitment.appAttributes)
```

Comment:
It might be worth having a decoder lookup defined somewhere, so that this could be
`fromAppAttributes = decoder[fromCommitment.channel.channelType](fromCommitment.appAttributes)`

Reply:
Would there be a case where we expect the wallet to decode anything other than consensus app attributes? Unless the wallet is is provided a decode function from an app, it should only every try this on the consensus rules.

If that's the case I think it's nicer to have it inlined so you understand the exact behaviour from looking at this function.

Reply:
True, good point!